From b76bb7e1bce4cb30f6b4b625334ce011b5a1654f Mon Sep 17 00:00:00 2001
From: Mathias Palm <mathias.palm@gmx.net>
Date: Sun, 9 May 2010 00:37:36 +0200
Subject: Copy common/types.h to vformat/vf_iface.h

Copy the content of common/types.h into vformat/vf_iface.h.
Keep common/types.h because some files depends on it.

Origin: vendor
Forwarded: http://vformat.cvs.sourceforge.net/viewvc/vformat/build/vformat/linux/ChangeLog?revision=1.3&view=markup
Last-Update: 2010-05-09
Applied-Upstream: http://vformat.cvs.sourceforge.net/viewvc/vformat/build/vformat/linux/ChangeLog?revision=1.3&view=markup
---
 ChangeLog                |  7 +++++++
 Makefile.am              |  1 -
 configure.ac             |  2 --
 src/vf_access.c          |  1 -
 src/vf_access_calendar.c |  2 --
 src/vf_access_wrappers.c |  1 -
 src/vf_create_object.c   |  1 -
 src/vf_delete.c          |  1 -
 src/vf_malloc.c          |  1 -
 src/vf_malloc_stdlib.c   |  2 --
 src/vf_modified.c        |  1 -
 src/vf_parser.c          |  1 -
 src/vf_reader.c          |  1 -
 src/vf_search.c          |  1 -
 src/vf_string_arrays.c   |  1 -
 src/vf_strings.c         |  1 -
 src/vf_writer.c          |  1 -
 vformat/vf_iface.h       | 45 ++++++++++++++++++++++++++++++----------
 18 files changed, 41 insertions(+), 30 deletions(-)

diff --git a/ChangeLog b/ChangeLog
index 4fe7df9..407796e 100644
--- a/ChangeLog
+++ b/ChangeLog
@@ -1,3 +1,10 @@
+2003-04-10    <mathias@ozone2.iup.uni-bremen.de>
+
+	* Moved the content of common/types.h in vf_iface.h,
+	common/types.h is kept because of depends of sources on it.
+
+	* included manpages and started info-files
+
 2002-11-22  Mathias Palm  <mathias.palm@gmx.net>
 
 	* manpages are generated from the header file and installed
diff --git a/Makefile.am b/Makefile.am
index 9ea4543..2bfe1c6 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -1,4 +1,3 @@
 SUBDIRS = src vformat test doc
 
-EXTRA_DIST = common/types.h vf_iface.h
 
diff --git a/configure.ac b/configure.ac
index d1b1041..744f321 100644
--- a/configure.ac
+++ b/configure.ac
@@ -1,7 +1,5 @@
 # Process this file with autoconf to produce a configure script.
 AC_INIT(libvformat, 1.13, Nick <Tilda@indigo8.freeserve.co.uk>)
-AC_CONFIG_SRCDIR([vf_iface.h])
-#AC_CONFIG_HEADER([config.h])
 
 AM_INIT_AUTOMAKE(libvformat, 1.13)
 AM_CONFIG_HEADER(config.h)
diff --git a/src/vf_access.c b/src/vf_access.c
index 7504150..5e28765 100644
--- a/src/vf_access.c
+++ b/src/vf_access.c
@@ -102,7 +102,6 @@ static const char vf_access_c_vss_id[] = "$Header: /cvsroot/vformat/src/vformat/
  ANSI C & System-wide Header Files
  *============================================================================*/
 
-#include <common/types.h>
 
 /*===========================================================================*
  Interface Header Files
diff --git a/src/vf_access_calendar.c b/src/vf_access_calendar.c
index 159c432..dbc8b74 100644
--- a/src/vf_access_calendar.c
+++ b/src/vf_access_calendar.c
@@ -52,8 +52,6 @@ static const char vf_access_calendar_c_vss_id[] = "$Header: /cvsroot/vformat/src
  ANSI C & System-wide Header Files
  *============================================================================*/
 
-#include <common/types.h>
-
 #include <time.h>
 #include <stdio.h>
 #include <ctype.h>
diff --git a/src/vf_access_wrappers.c b/src/vf_access_wrappers.c
index 2eea457..ac889dc 100644
--- a/src/vf_access_wrappers.c
+++ b/src/vf_access_wrappers.c
@@ -79,7 +79,6 @@ static const char vf_access_wrappers_c_vss_id[] = "$Header: /cvsroot/vformat/src
  ANSI C & System-wide Header Files
  *============================================================================*/
 
-#include <common/types.h>
 
 /*===========================================================================*
  Interface Header Files
diff --git a/src/vf_create_object.c b/src/vf_create_object.c
index ab540b9..2dc4376 100644
--- a/src/vf_create_object.c
+++ b/src/vf_create_object.c
@@ -54,7 +54,6 @@ static const char vf_create_object_c_vss_id[] = "$Header: /cvsroot/vformat/src/v
  ANSI C & System-wide Header Files
  *===========================================================================*/
 
-#include <common/types.h>
 
 /*===========================================================================*
  Interface Header Files
diff --git a/src/vf_delete.c b/src/vf_delete.c
index d70b8e4..49fc9a4 100644
--- a/src/vf_delete.c
+++ b/src/vf_delete.c
@@ -70,7 +70,6 @@ static const char vf_delete_c_vss_id[] = "$Header: /cvsroot/vformat/src/vformat/
  ANSI C & System-wide Header Files
  *=============================================================================*/
 
-#include <common/types.h>
 
 /*============================================================================*
  Interface Header Files
diff --git a/src/vf_malloc.c b/src/vf_malloc.c
index ef405d9..4649a5f 100644
--- a/src/vf_malloc.c
+++ b/src/vf_malloc.c
@@ -64,7 +64,6 @@ static const char vf_malloc_c_vss_id[] = "$Header: /cvsroot/vformat/src/vformat/
  ANSI C & System-wide Header Files
  *============================================================================*/
 
-#include <common/types.h>
 
 /*============================================================================*
  Interface Header Files
diff --git a/src/vf_malloc_stdlib.c b/src/vf_malloc_stdlib.c
index 6318fa5..b592986 100644
--- a/src/vf_malloc_stdlib.c
+++ b/src/vf_malloc_stdlib.c
@@ -38,8 +38,6 @@ static const char vf_malloc_stdlib_c_vss_id[] = "$Header: /cvsroot/vformat/src/v
  ANSI C & System-wide Header Files
  *============================================================================*/
 
-#include <common/types.h>
-
 #include <stdlib.h>
 #include <stdio.h>
 
diff --git a/src/vf_modified.c b/src/vf_modified.c
index 427698a..7cd1a5a 100644
--- a/src/vf_modified.c
+++ b/src/vf_modified.c
@@ -43,7 +43,6 @@ static const char vf_modified_c_vss_id[] = "$Header: /cvsroot/vformat/src/vforma
  ANSI C & System-wide Header Files
  *============================================================================*/
 
-#include <common/types.h>
 
 /*============================================================================*
  Interface Header Files
diff --git a/src/vf_parser.c b/src/vf_parser.c
index bf7b789..2c7221c 100644
--- a/src/vf_parser.c
+++ b/src/vf_parser.c
@@ -94,7 +94,6 @@ static const char vf_parser_c_vss_id[] = "$Header: /cvsroot/vformat/src/vformat/
  ANSI C & System-wide Header Files
  *===========================================================================*/
 
-#include <common/types.h>
 
 /*============================================================================*
  Interface Header Files
diff --git a/src/vf_reader.c b/src/vf_reader.c
index 667279f..a48246c 100644
--- a/src/vf_reader.c
+++ b/src/vf_reader.c
@@ -84,7 +84,6 @@ static const char vf_reader_c_vss_id[] = "$Header: /cvsroot/vformat/src/vformat/
  ANSI C & System-wide Header Files
  *=============================================================================*/
 
-#include <common/types.h>
 
 #include <stdio.h>
 #include <sys/stat.h>
diff --git a/src/vf_search.c b/src/vf_search.c
index 71be484..e303eae 100644
--- a/src/vf_search.c
+++ b/src/vf_search.c
@@ -64,7 +64,6 @@ static const char vf_search_c_vss_id[] = "$Header: /cvsroot/vformat/src/vformat/
  ANSI C & System-wide Header Files
  *============================================================================*/
 
-#include <common/types.h>
 
 /*===========================================================================*
  Interface Header Files
diff --git a/src/vf_string_arrays.c b/src/vf_string_arrays.c
index eee27f4..3a5addd 100644
--- a/src/vf_string_arrays.c
+++ b/src/vf_string_arrays.c
@@ -43,7 +43,6 @@ static const char vf_string_arrays_c_vss_id[] = "$Header: /cvsroot/vformat/src/v
  ANSI C & System-wide Header Files
  *=============================================================================*/
 
-#include <common/types.h>
 
 /*============================================================================*
  Interface Header Files
diff --git a/src/vf_strings.c b/src/vf_strings.c
index 204a188..05a32fa 100644
--- a/src/vf_strings.c
+++ b/src/vf_strings.c
@@ -60,7 +60,6 @@ static const char vf_strings_c_vss_id[] = "$Header: /cvsroot/vformat/src/vformat
  ANSI C & System-wide Header Files
  *=============================================================================*/
 
-#include <common/types.h>
 
 #include <ctype.h>
 #include <string.h>
diff --git a/src/vf_writer.c b/src/vf_writer.c
index ef3298a..29b83b5 100644
--- a/src/vf_writer.c
+++ b/src/vf_writer.c
@@ -68,7 +68,6 @@ static const char vf_writer_c_vss_id[] = "$Header: /cvsroot/vformat/src/vformat/
  ANSI C & System-wide Header Files
  *=============================================================================*/
 
-#include <common/types.h>
 
 #include <stdio.h>
 #include <ctype.h>
diff --git a/vformat/vf_iface.h b/vformat/vf_iface.h
index 8dc1b62..ab5e8d9 100644
--- a/vformat/vf_iface.h
+++ b/vformat/vf_iface.h
@@ -28,7 +28,7 @@
 
 FILE
     $Workfile: vf_iface.h $
-    $Revision: 1.30 $
+    $Revision: 1.31 $
     $Author: monos $
 
 ORIGINAL AUTHOR
@@ -42,6 +42,9 @@ REFERENCES
 
 MODIFICATION HISTORY
  *  $Log: vf_iface.h,v $
+ *  Revision 1.31  2003/04/10 12:49:16  monos
+ *  included type defs in vf_iface.h, started info documentation
+ *
  *  Revision 1.30  2002/12/07 17:25:44  monos
  *  *** empty log message ***
  *
@@ -135,28 +138,47 @@ MODIFICATION HISTORY
 #define _VF_IFACE_H_
 
 #ifndef NORCSID
-static const char vf_iface_h_vss_id[] = "$Header: /cvsroot/vformat/src/vformat/vf_iface.h,v 1.30 2002/12/07 17:25:44 monos Exp $";
+static const char vf_iface_h_vss_id[] = "$Header: /cvsroot/vformat/src/vformat/vf_iface.h,v 1.31 2003/04/10 12:49:16 monos Exp $";
 #endif
 
 /*============================================================================*
  Public Includes
  *============================================================================*/
+#include <stdarg.h>
+#include <time.h>
+
+/*============================================================================*
+ Public defines
+ *============================================================================*/
 
-#include <common/types.h>
+#undef  NULL
+#define NULL    0
 
-#include <stdarg.h>
+#undef  FALSE
+#define FALSE   ((bool_t)(0))
 
-/* c2man (for producing man pages) can't handle some lines in time.h */
+#undef  TRUE
+#define TRUE    ((bool_t)(1))
 
-#ifndef __C2MAN__
-#include <time.h>
+
+/*
+ * Basic types.
+ */
+#if defined(HAVE_INTTYPES_H)
+#include <inttypes.h>
 #else
-#define time_t long int
+typedef unsigned char       uint8_t;
+typedef unsigned short int  uint16_t;
+typedef unsigned int		uint32_t;
 #endif
 
-/*============================================================================*
- Public defines
- *============================================================================*/
+/*
+ * Please avoide 'BOOL' (from Windows) and 'bool' (from C++) - vformat is used
+ * extensively in pure C embedded applications where neither are relevant.
+ */
+typedef unsigned char bool_t;
+
+
 
 #if defined(__cplusplus)
 extern "C" {
@@ -434,6 +456,7 @@ extern "C" {
  Public Types
  *============================================================================*/
 
+
 /*
  * Type representing "parser" - an object used to parse VOBJECTS.
  */
